tensorflow/contrib/ignite/kernels/ignite_dataset_ops.cc

```diff
@@ -0,0 +1,147 @@
+
+    if (env_cache_name)
+      cache_name = std::string(env_cache_name);
+    else
+      OP_REQUIRES_OK(ctx, ParseScalarArgument<std::string>(ctx, "cache_name",
+                                                           &cache_name));
+
+    if (env_host)
+      host = std::string(env_host);
+    else
+      OP_REQUIRES_OK(ctx, ParseScalarArgument<std::string>(ctx, "host", &host));
+
+    if (env_port)
+      port = atoi(env_port);
+    else
+      OP_REQUIRES_OK(ctx, ParseScalarArgument<int32>(ctx, "port", &port));
+
+    if (env_local)
+      local = true;
+    else
+      OP_REQUIRES_OK(ctx, ParseScalarArgument<bool>(ctx, "local", &local));
+
+    if (env_part)
+      part = atoi(env_part);
+    else
+      OP_REQUIRES_OK(ctx, ParseScalarArgument<int32>(ctx, "part", &part));
+
+    if (env_page_size)
+      page_size = atoi(env_page_size);
+    else
+      OP_REQUIRES_OK(ctx,
+                     ParseScalarArgument<int32>(ctx, "page_size", &page_size));
+
+    if (env_username)
+      username = std::string(env_username);
+    else
+      OP_REQUIRES_OK(
+          ctx, ParseScalarArgument<std::string>(ctx, "username", &username));
+
+    if (env_password)
+      password = std::string(env_password);
+    else
+      OP_REQUIRES_OK(
+          ctx, ParseScalarArgument<std::string>(ctx, "password", &password));
+
+    if (env_certfile)
+      certfile = std::string(env_certfile);
+    else
+      OP_REQUIRES_OK(
+          ctx, ParseScalarArgument<std::string>(ctx, "certfile", &certfile));
+
+    if (env_keyfile)
+      keyfile = std::string(env_keyfile);
+    else
+      OP_REQUIRES_OK(
+          ctx, ParseScalarArgument<std::string>(ctx, "keyfile", &keyfile));
+
+    if (env_cert_password)
+      cert_password = std::string(env_cert_password);
+    else
+      OP_REQUIRES_OK(ctx, ParseScalarArgument<std::string>(ctx, "cert_password",
+                                                           &cert_password));
+
+    const Tensor* schema_tensor;
+    OP_REQUIRES_OK(ctx, ctx->input("schema", &schema_tensor));
+    OP_REQUIRES(ctx, schema_tensor->dims() == 1,
+                errors::InvalidArgument("`schema` must be a vector."));
+
+    std::vector<int32> schema;
+    schema.reserve(schema_tensor->NumElements());
+    for (int i = 0; i < schema_tensor->NumElements(); i++) {
+      schema.push_back(schema_tensor->flat<int32>()(i));
+    }
+
+    const Tensor* permutation_tensor;
+    OP_REQUIRES_OK(ctx, ctx->input("permutation", &permutation_tensor));
+    OP_REQUIRES(ctx, schema_tensor->dims() == 1,
+                errors::InvalidArgument("`permutation` must be a vector."));
+
+    std::vector<int32> permutation;
+    permutation.reserve(permutation_tensor->NumElements());
+    for (int i = 0; i < permutation_tensor->NumElements(); i++) {
+      permutation.push_back(permutation_tensor->flat<int32>()(i));
+    }
+
+    *output =
+        new IgniteDataset(ctx, cache_name, host, port, local, part, page_size,
```

Comment:
Use `std::move` for all the non-POD types passed by value here.

Reply:
Done.